tests/unit/worker/test_submission_worker.py

```diff
@@ -220,3 +224,53 @@ def test_get_or_create_sqs_queue_for_non_existing_queue(self):
         queue_url = self.sqs_client.get_queue_url(QueueName='test_queue_2')['QueueUrl']
         self.assertTrue(queue_url)
         self.sqs_client.delete_queue(QueueUrl=queue_url)
+
+
+class DownloadAndExtractZipFileTest(BaseAPITestClass):
+    def setUp(self):
+        super(DownloadAndExtractZipFileTest, self).setUp()
+        self.zip_name = "test"
+        self.req_url = "{}/{}".format(self.testserver, self.zip_name)
+        self.extract_location = join(self.BASE_TEMP_DIR, "test-dir")
+        self.download_location = join(self.extract_location, "{}.zip".format(self.zip_name))
+        create_dir(self.extract_location)
+
+        self.file_name = "test_file.txt"
+        self.file_content = b"file_content"
+
+        self.zip_file = BytesIO()
+        with zipfile.ZipFile(self.zip_file, mode="w", compression=zipfile.ZIP_DEFLATED) as zipper:
+            zipper.writestr(self.file_name, self.file_content)
+
+    def tearDown(self):
+        if os.path.exists(self.extract_location):
+            shutil.rmtree(self.extract_location)
+
+    @responses.activate
+    @mock.patch("scripts.workers.submission_worker.delete_zip_file")
+    @mock.patch("scripts.workers.submission_worker.extract_zip_file")
```

Comment:
@nikochiko since this is creating the travis issue maybe you should move this as well to #2532

Reply:
@yashdusing I don't think this change will go with the purpose of the PRs as the main objective of this PR is to write these tests. As the other PR is approved, I think we can wait for its merge. 
Even if we edit the tests to work, it will be very temporary or the tests will be very sensitive to the specific modules used.
What do you think?